frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java

```diff
@@ -0,0 +1,324 @@
+package com.virtuslab.gitmachete.frontend.actions.pushdialog;
+
+import java.awt.Component;
+import java.awt.event.ActionEvent;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.JRootPane;
+import javax.swing.border.Border;
+
+import com.intellij.dvcs.push.PrePushHandler;
+import com.intellij.dvcs.push.PushController;
+import com.intellij.dvcs.push.PushInfo;
+import com.intellij.dvcs.push.PushSource;
+import com.intellij.dvcs.push.PushSupport;
+import com.intellij.dvcs.push.PushTarget;
+import com.intellij.dvcs.push.VcsPushOptionValue;
+import com.intellij.dvcs.push.ui.PushLog;
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.dvcs.push.ui.VcsPushUi;
+import com.intellij.dvcs.repo.Repository;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.util.registry.Registry;
+import com.intellij.util.ui.JBDimension;
+import com.intellij.util.ui.JBUI;
+import com.intellij.util.ui.UIUtil;
+import com.intellij.util.ui.components.BorderLayoutPanel;
+import io.vavr.collection.List;
+import net.miginfocom.swing.MigLayout;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.initialization.qual.UnderInitialization;
+import org.checkerframework.checker.initialization.qual.UnknownInitialization;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public final class GitPushDialog extends DialogWrapper implements VcsPushUi {
+  private static final String DIMENSION_KEY = "Vcs.Push.Dialog.v2";
+
+  private static final int CENTER_PANEL_HEIGHT = 450;
+  private static final int CENTER_PANEL_WIDTH = 800;
+
+  private final Project project;
+  private final PushLog listPanel;
+  private final PushController pushController;
+  private final Action pushAction;
+  private final boolean isForcePushRequired;
+
+  @UIEffect
+  // Needed for the method MyVcsPushDialog#getPushController which can be run safely
+  // because the push controller is fully created within VcsPushDialog constructor.
+  @SuppressWarnings("nullness:method.invocation.invalid")
```

Comment:
To reduce the range of suppression, pls apply the following changes:

```diff
diff --git a/config/checkstyle/tree_walker_module_directives.xml b/config/checkstyle/tree_walker_module_directives.xml
index 7cc1eca2..8b129647 100644
--- a/config/checkstyle/tree_walker_module_directives.xml
+++ b/config/checkstyle/tree_walker_module_directives.xml
@@ -160,6 +160,12 @@
     <property name="format" value="\bsynchronized\b"/>
     <property name="message" value="Avoid lock-based synchronization, consider io.vavr.Lazy, Task.Backgroundable and/or GuiUtils.invokeLaterIfNeeded(...) instead"/>
 </module>
+<module name="Regexp">
+    <property name="illegalPattern" value="true"/>
+    <property name="ignoreComments" value="true"/>
+    <property name="format" value="\([^@,)/*]+ this[,)]"/>
+    <property name="message" value="Unannotated (and therefore likely redundant) explicit receiver"/>
+</module>
 <module name="Regexp">
     <property name="illegalPattern" value="true"/>
     <property name="ignoreComments" value="true"/>
diff --git a/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java b/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java
index 4d4719fc..741cccc7 100644
--- a/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java
+++ b/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java
@@ -37,6 +37,7 @@ import com.intellij.util.ui.components.BorderLayoutPanel;
 import io.vavr.collection.List;
 import net.miginfocom.swing.MigLayout;
 import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.initialization.qual.NotOnlyInitialized;
 import org.checkerframework.checker.initialization.qual.UnderInitialization;
 import org.checkerframework.checker.initialization.qual.UnknownInitialization;
 import org.checkerframework.checker.nullness.qual.Nullable;
@@ -56,7 +57,6 @@ public final class GitPushDialog extends DialogWrapper implements VcsPushUi {
   @UIEffect
   // Needed for the method MyVcsPushDialog#getPushController which can be run safely
   // because the push controller is fully created within VcsPushDialog constructor.
-  @SuppressWarnings("nullness:method.invocation.invalid")
   public GitPushDialog(
       Project project,
       List<? extends Repository> selectedRepositories,
@@ -83,7 +83,7 @@ public final class GitPushDialog extends DialogWrapper implements VcsPushUi {
 
   private static final class MyVcsPushDialog extends VcsPushDialog {
 
-    private final @UnknownInitialization GitPushDialog dialog;
+    private final @NotOnlyInitialized GitPushDialog dialog;
 
     @UIEffect
     MyVcsPushDialog(
@@ -98,10 +98,17 @@ public final class GitPushDialog extends DialogWrapper implements VcsPushUi {
       this.dialog = dialog;
     }
 
-    public PushController getPushController() {
+    public PushController getPushController(@UnknownInitialization(VcsPushDialog.class) MyVcsPushDialog this) {
       return myController;
     }
 
+    @Override
+    @UIEffect
+    @SuppressWarnings("nullness:method.invocation.invalid")
+    public void updateOkActions(@UnknownInitialization(VcsPushDialog.class) MyVcsPushDialog this) {
+      super.updateOkActions();
+    }
+
     @Override
     @UIEffect
     public @Nullable VcsPushOptionValue getAdditionalOptionValue(PushSupport support) {
```

Reply:
(plus remove redundant explicit receivers, as per the patched Checkstyle above)

Reply:
thanks!